src/psyclone/tests/psyir/frontend/fparser2_do_handler_test.py

```diff
@@ -255,3 +259,71 @@ def test_do_inside_while(fortran_reader, fortran_writer, tmpdir):
 '''
     assert correct == output
     assert Compile(tmpdir).string_compiles(output)
+
+
+def test_do_concurrent(fortran_reader, fortran_writer, tmpdir):
+    '''Check that the loop handler correctly identifies do concurrent loops '''
+
+    # Single do concurrent loop
+    code = '''
+    subroutine test
+      integer, parameter :: N =10, M=20
+      integer, dimension(N,M) :: array
+      integer i
+
+      do concurrent(i=1:N)
+        array(i,5) = 1
+      end do
+    end subroutine test
+    '''
+    psyir = fortran_reader.psyir_from_source(code)
+    loops = psyir.walk(Loop)
+    assert len(loops) == 1
+    assert loops[0].variable.name == "i"
+    assert loops[0].start_expr == Literal("1", INTEGER_TYPE)
+    assert isinstance(loops[0].stop_expr, Reference)
+    assert loops[0].stop_expr.symbol.name == "n"
+    assert loops[0].step_expr == Literal("1", INTEGER_TYPE)
+
+    # Do concurrent with nested loops and step values
+    code = '''
+    subroutine test
+      integer, parameter :: N =10, M=20
+      integer, dimension(N,M) :: array
+      integer i, j
+
+      do concurrent(i=1:N:3, j=1:M:2)
```

Comment:
Can a `do concurrent` itself contain a nested, explicit loop? e.g.

    do concurrent(i=1:N)
      do j=1,10
         a(i,j) = 1.0
      end do
    end do

Please add a test for this if so. (Also, when `do concurrent` is nested inside another loop of a different kind.)

Reply:
Done